elastic/elastic-agent
# Add complete flag environment variable to agent metadata

## What does this PR do?



Closes https://github.com/elastic/fleet-server/issues/1754. Add environment variable to agent complete image and add it to agent metada.

## Why is it important?

We need a mechanism to detect if the agent enrolled is a `-complete` image in Synthetics to show early warnings which the spec does not provide.

## Checklist




## How to test this PR locally


1. Build agent docker images locally.
2. Enroll both an `elastic-agent` and `elastic-agent-complete` containers on a policy.
3. Check both agents metadata for `complete` flag:

```
//complete
 "local_metadata": {
    "elastic": {
        ...,
        "complete": true,
      }
    },
}
//vs non-complete
 "local_metadata": {
    "elastic": {
      "agent": {
        ...,
        "complete": false,
      }
    },
}
```
